util/offline_db/get_resources_from_db.py

```diff
@@ -0,0 +1,181 @@
+# contributors may be used to endorse or promote products derived from
+# this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+# A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+# OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+"""
+This is a script to get all resources from the database and save them to a json file
+and download all resources to a directory. The resources in the resources.json file
+that is generated will be updated to point to the local resources instead of the
+database resources. This script is used to create the offline database.
+To use the resources.json with gem5 you need to update the gem5 config to point to the
+resources.json file.
+"""
+
+import argparse
+import json
+from urllib import request, parse
+import pathlib
+
+
+def cli():
+    parser = argparse.ArgumentParser(
+        description="Get resources from the database"
+    )
+    parser.add_argument(
+        "--api_key",
+        type=str,
+        default="OIi5bAP7xxIGK782t8ZoiD2BkBGEzMdX3upChf9zdCxHSnMoiTnjI22Yw5kOSgy9",
+        help="API key to access the database",
+    )
+    parser.add_argument(
+        "--db_name",
+        type=str,
+        default="gem5-vision",
+        help="Name of the database",
+    )
+    parser.add_argument(
+        "--collection_name",
+        type=str,
+        default="resources",
+        help="Name of the collection",
+    )
+    parser.add_argument(
+        "--auth_url",
+        default="https://realm.mongodb.com/api/client/v2.0/app/data-ejhjf/auth/providers/api-key/login",
+        type=str,
+        help="Authentication URL",
+    )
+    parser.add_argument(
+        "--db_url",
+        type=str,
+        default="https://data.mongodb-api.com/app/data-ejhjf/endpoint/data/v1",
+        help="Database URL",
+    )
+    parser.add_argument(
+        "--data_source",
+        type=str,
+        default="gem5-vision",
+        help="Data source name",
+    )
+    parser.add_argument(
+        "--output_dir",
+        type=str,
+        help="Output directory absolute path, default is the cwd. The resources.json and all resources will be saved in this directory",
+    )
+    args = parser.parse_args()
+
+    if args.output_dir == None:
+        output_path = pathlib.Path(pathlib.Path.cwd())
```

Comment:
```suggestion
        output_path = pathlib.Path.cwd()
```

I think this is equivalent